pkg/parse/root.go

```diff
@@ -729,20 +729,20 @@ func prependRootSyncRemediatorStatus(ctx context.Context, c client.Client, syncN
 		return status.APIServerError(err, "failed to get RootSync: "+syncName)
 	}
 
+	// Skip updating if already reported by this reconciler or the remote reconciler (inverted)
 	var errs []v1beta1.ConfigSyncError
 	for _, conflictErr := range conflictErrs {
-		conflictCSEError := conflictErr.ToCSE()
-		conflictPairCSEError := conflictErr.CurrentManagerError().ToCSE()
+		cse := conflictErr.ToCSE()
+		invertedCSE := conflictErr.Invert().ToCSE()
 		errorFound := false
 		for _, e := range rs.Status.Sync.Errors {
-			// Dedup the same remediator conflict error.
-			if e.Code == status.ManagementConflictErrorCode && (e.ErrorMessage == conflictCSEError.ErrorMessage || e.ErrorMessage == conflictPairCSEError.ErrorMessage) {
+			if e.Code == status.ManagementConflictErrorCode && (e.ErrorMessage == cse.ErrorMessage || e.ErrorMessage == invertedCSE.ErrorMessage) {
```

Comment:
Are we lacking test coverage for this? I'm surprised the tests were passing with the previous bugged implementation

Reply:
TestConflictingDefinitions_RootToRoot is the only test covering remote conflict status reporting. The way those task group predicates are written it passes whether the conflict was reported by the local or remote reconciler. That's what was causing the test to fail in https://github.com/GoogleContainerTools/kpt-config-sync/pull/1343

Also, those predicates pass regardless of how many errors are reported. So it doesn't strictly require de-duping either.

Reply:
I considered changing the predicates to require exactly one conflict error, but that won't work either, because they will see exactly one error before they see two.

I'm not really sure how to prove the negative that it only ever sees one.

Wanna backfill some unit tests for me?

Reply:
https://github.com/GoogleContainerTools/kpt-config-sync/pull/1345

Reply:
Thanks for the tests!
They caught a subtle bug in my Invert method, and I figured out why the message for KptManagementConflictError was weird (left a comment).
All but one of the unexpected test behaviors have been fixed now. The other I will leave for another time.

Reply:
Glad it helped! I was also surprised by some of the existing behavior when I wrote the test.

It also helps with understanding the before/after behavior with this change. Maybe lets create a ticket for the other unexpected behavior